crates/vm/levm/src/precompiles.rs

```diff
@@ -501,12 +501,186 @@ pub fn ecmul(
+    for group_number in 0..groups_number {
+        // Define the group indexes and slice calldata to get the group data
+        let group_start = group_number
+            .checked_mul(192)
+            .ok_or(InternalError::ArithmeticOperationOverflow)?;
+        let next_group_start = (group_number
+            .checked_add(1)
+            .ok_or(InternalError::ArithmeticOperationOverflow)?)
+        .checked_mul(192)
+        .ok_or(InternalError::ArithmeticOperationOverflow)?;
+
+        let group_data = calldata
+            .get(group_start..next_group_start)
+            .ok_or(InternalError::SlicingError)?;
+
+        let first_point_x = group_data.get(..32).ok_or(InternalError::SlicingError)?;
+        let first_point_y = group_data.get(32..64).ok_or(InternalError::SlicingError)?;
+
+        // Infinite is defined by (0,0). Any other zero-combination is invalid
+        if (U256::from_big_endian(first_point_x) == U256::zero()
+            && U256::from_big_endian(first_point_y) != U256::zero())
+            || (U256::from_big_endian(first_point_x) != U256::zero()
+                && U256::from_big_endian(first_point_y) == U256::zero())
+        {
+            return Err(VMError::PrecompileError(PrecompileError::DefaultError));
+        }
+
+        let first_point_y = BN254FieldElement::from_bytes_be(first_point_y)
+            .map_err(|_| PrecompileError::DefaultError)?;
+        let first_point_x = BN254FieldElement::from_bytes_be(first_point_x)
+            .map_err(|_| PrecompileError::DefaultError)?;
+
+        let second_point_x_first_part =
+            group_data.get(96..128).ok_or(InternalError::SlicingError)?;
+        let second_point_x_second_part =
+            group_data.get(64..96).ok_or(InternalError::SlicingError)?;
+
+        // Infinite is defined by (0,0). Any other zero-combination is invalid
+        if (U256::from_big_endian(second_point_x_first_part) == U256::zero()
+            && U256::from_big_endian(second_point_x_second_part) != U256::zero())
+            || (U256::from_big_endian(second_point_x_first_part) != U256::zero()
+                && U256::from_big_endian(second_point_x_second_part) == U256::zero())
+        {
+            return Err(VMError::PrecompileError(PrecompileError::DefaultError));
+        }
+
+        let second_point_y_first_part = group_data
+            .get(160..192)
+            .ok_or(InternalError::SlicingError)?;
+        let second_point_y_second_part = group_data
+            .get(128..160)
+            .ok_or(InternalError::SlicingError)?;
+
+        // Infinite is defined by (0,0). Any other zero-combination is invalid
+        if (U256::from_big_endian(second_point_y_first_part) == U256::zero()
+            && U256::from_big_endian(second_point_y_second_part) != U256::zero())
+            || (U256::from_big_endian(second_point_y_first_part) != U256::zero()
+                && U256::from_big_endian(second_point_y_second_part) == U256::zero())
```

Comment:
ditto

Reply:
Done in [Use xor in non valid infinity checking](https://github.com/lambdaclass/ethrex/pull/1560/commits/0f72780ba6d2203ae890f3956026011b7a8386e5)